[quant][graphmode][fix] preserve the type of original value when inserting dequant node

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* #35077 [quant][graphmode] Fold quantized prepacking ops
* #35073 [quant] Make conv2d_prepack and linear_prepack pure
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* #34572 [quant][graphmode] quantization support for aten::add
* #34518 [quant][graphmode] quantization work for prim::If
* #34571 [quant][graphmode][fix] use observed_values_ to check values are observed
* #34414 [quant][graphmode][refactor] insertObservers for Block
* #34411 [quant][graphmode][fix] Insert dequantize before use node
* **#34349 [quant][graphmode][fix] preserve the type of original value when inserting dequant node**

Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

jerryzh168 committed:
[quant][graphmode][fix] preserve the type of original value when inserting dequant node

Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -1140,7 +1140,8 @@ void insertDeQuantCall(Graph* graph,
       graph->create(Symbol::aten("dequantize"), {quantized_val});
     dequant->output()->setDebugName(
-        original_val->debugName() + ".dequant." + c10::guts::to_string(i));
+        original_val->debugName() + ".dequant." + c10::guts::to_string(i))
```
If this was an issue that needed a fix, suggest having a simple test that checks for the type

jerryzh168 replied:
this actually only can happen when we do swap dequantize for list of tensors, and it will be tested later in the stack

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar reviewed (approved):
LGTM, assuming this is tested later.

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] preserve the type of original value when inserting dequant node"


Summary:
Set the output type of dequantize node to the type of original value
this is to fix swap dequantize tensor list

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504456](https://our.internmc.facebook.com/intern/diff/D20504456)

[ghstack-poisoned]
